v2.0

alirezaaa opened the pull request:
I changed lots of things with improves its code reviewing, performance, supporting and etc.

DeveloperPaul123 commented:
Can you be more specific in what you changed exactly. Also could you post some screen shots?

Thanks

alirezaaa commented:
Sure my friend.

- I updated the Gradle to v2.10
- `.gitignore` file updated with better and more statements from [gitignore.io](gitignore.io)
- Packages names are now based on your GitHub account.
- Cleaned up using Android Studio
- The library name changed to 'library' and also the sample name changed to 'app'
- `targetSdkVersion` updated to v23.0.2 and also all Android dependencies are updated.
- Unnecessary Android `CardView` library removed.
- `com.android.tools.build:gradle` updated to v1.5.0
- Android support `FloatingActionButton` used for the material theme.
- `ApplicationId` changed to `com.github.developerpaul123.filepickerlibrary`
- Some bug fixes
- `AppCompatActivity` used instead of `Activity` class.
- `AppCompat` related theme is used.
- Unnecessary `AppTheme.NoActionBar` style removed.
- repositories updated and fixed.
- A new icon added to be used within the `FloatingActionButton`.
- A behavior added for the `FloatingActionButton` (`ScrollAwareFABBehavior `)
- Native Android `SnackBar` used instead of deprecated `SnackBar` Builder library.

**Note:** Giving screenshots isn't necessary. I tested on API 22 and API 17 and everything works fine.

**Note:** I removed all files and statements related to your maven. Sorry about that. you can put them back or try [jitpack.io](https://jitpack.io/).

DeveloperPaul123 commented:
Ok so no changes to how the actual library works or how you access it? 

I wanted screenshots so I can see what it looks like now. 

And no problem with the maven stuff, I've been using jitpack for the last 5 releases. 
Thanks

DeveloperPaul123 commented:
Also, did you happen to address any of the issues in this pull request?

Just looking through the code looks like you did a lot of work with many improvements! Thank you so much!

alirezaaa commented:
Yes, I didn't change anything that effects on its usage and how users access it. Users don't need to change anything actually.

It looks like what it was before, if you mean its layouts' design.

No, I didn't address any issues.

**Note:** Consider changing the `versionName` to v3.0 to prevent from conflicts between already released packages' version. I made a pull request for that.

You're welcome dear Paul. I like developers who care about their libraries and it makes me feel good to contribute :)

DeveloperPaul123 commented:
Thanks for everything. I'll go ahead and merge. I appreciate the kind words and I'll update the screen shots, maybe add a video too.
